### DISCUSSION TITLE
Issue events that trigger workflows

### DISCUSSION THREAD

**Discussion by robertp** [community/community#26867] [Actions]
<p>I think this belongs in the ‘Github Actions’ category, please fix if not.</p>
<p>I’m trying to understand the distinction between issue events as described in two places in the documentation:</p>
<p>Link 1 <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows" rel="noopener nofollow ugc">https://docs.github.com/en/actions/reference/events-that-trigger-workflows</a><br>
Link 2 <a href="https://docs.github.com/en/developers/webhooks-and-events/issue-event-types" rel="noopener nofollow ugc">https://docs.github.com/en/developers/webhooks-and-events/issue-event-types</a></p>
<p>In the first doc (‘Events that trigger workflows’), the following list of issue events is provided (Link 1):</p>
<ul>
<li><code>opened</code></li>
<li><code>edited</code></li>
<li><code>deleted</code></li>
<li><code>transferred</code></li>
<li><code>pinned</code></li>
<li><code>unpinned</code></li>
<li><code>closed</code></li>
<li><code>reopened</code></li>
<li><code>assigned</code></li>
<li><code>unassigned</code></li>
<li><code>labeled</code></li>
<li><code>unlabeled</code></li>
<li><code>locked</code></li>
<li><code>unlocked</code></li>
<li><code>milestoned</code></li>
<li><code>demilestoned</code></li>
</ul>
<p>In the second doc, a much longer list is provided.  Some of the events are in both lists. I could understand that specifically to make webhooks call out, only a subset of issue events is supported.  However, on the parent page of Link 2 (sorry, new posters can only use 2 URLs) there’s some prominent text: ‘Your integrations can subscribe and react to events on GitHub.’. The nav from that page takes you to the longer list… To me, this implies that the longer list is available for webhooks.</p>
<p>I’m probably just missing something, but any explanations would be helpful. Thank you!</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/robertp">@robertp</a> ,</p>
<blockquote>
<p>In the second doc, a much longer list is provided.</p>
</blockquote>
<p>The following description from the docs “<a href="https://docs.github.com/en/developers/webhooks-and-events/issue-event-types" rel="noopener nofollow ugc"><strong>Issue event types</strong></a>” (<strong>Link 2</strong>) can answer your doubt:</p>
<blockquote>
<p>Issue events are triggered by activity in issues and pull requests and are available in the <a href="https://docs.github.com/en/v3/issues/events" rel="noopener nofollow ugc">Issue Events API</a> and the <a href="https://docs.github.com/en/v3/issues/timeline" rel="noopener nofollow ugc">Timeline Events API</a>.</p>
</blockquote>
<p>It includes some event types of pull request, for example, add, edit or delete comments. This is consistent with the <a href="https://docs.github.com/en/rest/reference/issues" rel="noopener nofollow ugc">issue API</a>.<br>
There is a point you need to know that, on GitHub, <strong>pull request</strong> is a type of <strong>issue</strong>.</p>
<br>
<p>On the docs “<a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows" rel="noopener nofollow ugc"><strong>Events that trigger workflows</strong></a>” ( <strong>Link 1</strong>):</p>
<ul>
<li>
<p>The <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issues" rel="noopener nofollow ugc"><strong>issues</strong></a> event only includes the activities occur on issue.</p>
</li>
<li>
<p>The <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request" rel="noopener nofollow ugc"><strong>pull_request</strong></a> event only includes the activities occur on pull request.</p>
</li>
<li>
<p>For the common event types, such as adding, editing or deleting comments, there are the corresponding separate events. For example, the <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issue_comment" rel="noopener nofollow ugc"><strong>issue_comment</strong></a> event, when you add, edit or delete comments on issues or pull requests, this event will be triggered.</p>
</li>
</ul>
<br>
<blockquote>
<p>on the parent page of Link 2 (sorry, new posters can only use 2 URLs) there’s some prominent text: ‘Your integrations can subscribe and react to events on GitHub.’.</p>
</blockquote>
<p>This is saying that sending event webhooks to the external services.<br>
GitHub can integrate many external services, GitHub provides the way to send webhooks to the external services when certain events happen.<br>
To set webhooks to notify the external services, you can follow the steps below:</p>
<ol>
<li>
<p><strong>Set repository level webhooks</strong><br>
Navigate to repository’s <strong>Settings</strong> &gt; <strong>Webhooks</strong>, click “Add webhook” button to add a new webhook, or edit or delete the existing webhooks.</p>
</li>
<li>
<p><strong>Set organization level webhooks</strong><br>
Navigate to organization’s <strong>Settings</strong> &gt; <strong>Webhooks</strong>.</p>
</li>
</ol>
<p>More details, see “<a href="https://docs.github.com/en/developers/webhooks-and-events/webhooks" rel="noopener nofollow ugc"><strong>Webhooks</strong></a>”.</p>

**Comment by robertp** [comment]
<p>Thank you for your answer. It’s not really what I’m after. Let me ask the question more directly.</p>
<p>Can this issues event:</p>
<p><a href="https://docs.github.com/en/developers/webhooks-and-events/issue-event-types#moved_columns_in_project" rel="noopener nofollow ugc">moved_columns_in_project</a></p>
<p>be used to trigger a GitHub Action?  For example:</p>
<pre><code class="lang-auto">name: move
on:
  issues:
    types: [moved_columns_in_project]
jobs:
  move:
    runs-on: ubuntu-latest
    steps:
      - name: Move it
        uses: ...etc
</code></pre>
<p>I know that for example the labeled issues event (in the list provided here: <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issues" rel="noopener nofollow ugc">https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issues</a>) <strong>can</strong> be used that way. Thanks!</p>

**Comment by robertp** [comment, marked as answer]
<p>Easy enough to figure out the answer I suppose - you can only use the events listed here: <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issues" rel="noopener nofollow ugc">https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issues</a> to trigger workflow events; for the other set of events, you have to use the issue events API to scrape them out, periodically or triggered by e.g. the edited event triggering a workflow (and the workflow scraping the API).</p>
